File: 351004/Shkredova/task310-rest/src/main/java/com/example/task310/mapper/CreatorMapper.java
package com.example.task310.mapper;

import com.example.task310.dto.CreatorRequestTo;
import com.example.task310.dto.CreatorResponseTo;
import com.example.task310.model.Creator;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

@Mapper(componentModel = "spring")
public interface CreatorMapper {
    @Mapping(target = "id", ignore = true)
    @Mapping(target = "created", ignore = true)
    @Mapping(target = "modified", ignore = true)
    @Mapping(target = "news", ignore = true)
    Creator toEntity(CreatorRequestTo request);

    CreatorResponseTo toResponse(Creator creator);
}

File: 351002/Abrashin/publisher/src/main/kotlin/com/example/distcomp/mapper/CreatorMapper.kt
package com.example.distcomp.mapper

import com.example.distcomp.dto.request.CreatorRequestTo
import com.example.distcomp.dto.response.CreatorResponseTo
import com.example.distcomp.model.Creator
import org.mapstruct.Mapper
import org.mapstruct.Mapping

@Mapper(componentModel = "spring")
interface CreatorMapper {
    @Mapping(target = "role", ignore = true)
    fun toEntity(request: CreatorRequestTo): Creator
    fun toResponse(entity: Creator): CreatorResponseTo
}
